contracts/Core/AssetManager.sol

```diff
@@ -106,31 +82,37 @@ contract AssetManager is ACL, AssetStorage, Constants, StateManager, IAssetManag
         onlyRole(ASSET_MODIFIER_ROLE)
         checkState(State.Confirm, parameters.epochLength())
     {
-        require(collections[id].assetType == uint8(AssetTypes.Collection), "Asset is not a collection");
+        require(id != 0, "ID cannot be 0");
+        require(collections[id].id == id, "Asset is not a collection");
 
         uint32 epoch = parameters.getEpoch();
         if (assetStatus) {
             if (!collections[id].active) {
                 activeAssets.push(id);
                 collections[id].assetIndex = uint8(activeAssets.length);
+                collections[id].active = assetStatus;
+                emit CollectionActivityStatus(collections[id].active, id, epoch, block.timestamp);
             }
         } else {
             if (collections[id].active) {
-                uint8 assetIndex = collections[id].assetIndex;
-                if (assetIndex == activeAssets.length) {
-                    activeAssets.pop();
-                } else {
-                    activeAssets[assetIndex - 1] = activeAssets[activeAssets.length - 1];
-                    collections[activeAssets[assetIndex - 1]].assetIndex = assetIndex;
-                    activeAssets.pop();
-                }
-                collections[id].assetIndex = 0;
+                pendingDeactivations.push(id);
             }
         }
+    }
 
-        collections[id].active = assetStatus;
-
+    function deactivateCollection(uint32 epoch, uint8 id) external override onlyRole(ASSET_CONFIRMER_ROLE) {
```

Comment:
Hey just a suggestion, 
having setCollectionStatus and deactivteCollection,is little ambigious with new code. 
If its not going to take not much time, and you agree. 
Can we have something like "addToPendingDeactivations" and "executePendingDeactivations" ?

Reply:
setCollectionStatus also deals with activation of collection so changing the name to addToPendingDeactivations doesn't make sense. We can change deactivateCollection to executePendingDeactivations though